### PR TITLE
Lance/pubs

### PR DESCRIPTION
**Pull request by LGG233**
@kylewhitaker - 
I've added functionality to add and remove publications to user profile. Welcome your thoughts. A couple of points to consider while reviewing:
- although I have `res.sendStatus()` on the server side to tell the user that they're trying to save a title already in their db (for example), when those are invoked no alert / message appears in the DOM. Shouldn't the user get some sort of signal when this happens? I thought that's what `sendStatus` did but maybe I'm not using it properly. 
- please let me know if you find that I overcomplicated my code. My solutions may not be elegant but they're effective (at least I think so...). Put another way: just because it works doesn't mean it shouldn't be improved. 
This is getting close! We should be able to start working with the Twitter API next. Thanks for all your help!

### CODE REVIEW

**Comment by LGG233** [comment]
OK I just found a problem... It looks like I used the same route name for adding publications to the publications database and adding publications to the user's personal db. Because when I try to add a new pub to the master list, it shows up in my profile. Sigh... So don't review yet!

**Comment by LGG233** [comment]
OK fixed! Phew

File: client/src/components/signup.jsx [hunk 1]
```diff
@@ -32,8 +32,8 @@ class SignUp extends Component {
 
   handleSubmit = async () => {
     try {
-      const { username, password, email } = this.state;
-      await AmplifyAuth.signUp({ username, password, attributes: { email } });
+      const { name, username, password, email } = this.state;
+      await AmplifyAuth.signUp({ name, username, password, attributes: { email } });
```
Comment (kylewhitaker) [inline client/src/components/signup.jsx]: I think username and password are the only “core” attributes for Amplify/Cognito sign up. `name` should be inside the `attributes` object, like `email`.

File: client/src/components/publications.jsx [hunk 2]
```diff
@@ -1,6 +1,29 @@
...
+      name: publication.name,
+      twitterHandle: publication.twitterHandle,
+    })
+      .then((response) => {
+        if (response.statusText === 'OK') {
+          this.props.getUser();
+          navigate('/userProfile');
+        }
+      })
+      .catch((error) => {
```
Comment (kylewhitaker) [inline client/src/components/publications.jsx:22]: This is where you would handle 409 errors. Any error code 4xx or 5xx will fall into the `catch`.

**Review by kylewhitaker — APPROVED** [review]
LGTM! Nice job!

File: client/src/components/publications.jsx (hunk 2 above)
Comment (LGG233) [reply client/src/components/publications.jsx:22]: I still don't seem to get it - let's go through on Zoom next time we talk... Thanks

**Pull request merged**